src/tests/sqlsmith/src/sql_gen/expr.rs

```diff
@@ -12,54 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::HashMap;
-use std::sync::LazyLock;
-
 use itertools::Itertools;
 use rand::seq::SliceRandom;
 use rand::Rng;
-use risingwave_common::types::DataTypeName;
 use risingwave_expr::expr::AggKind;
-use risingwave_frontend::expr::{
-    agg_func_sigs, cast_sigs, func_sigs, AggFuncSig, CastContext, CastSig, ExprType, FuncSign,
-};
+use risingwave_frontend::expr::{agg_func_sigs, cast_sigs, func_sigs, CastContext, ExprType};
 use risingwave_sqlparser::ast::{
     BinaryOperator, Expr, Function, FunctionArg, FunctionArgExpr, Ident, ObjectName,
     TrimWhereField, UnaryOperator, Value,
 };
 
-use crate::sql_gen::utils::data_type_name_to_ast_data_type;
+use crate::sql_gen::types::{DataType, AGG_FUNC_TABLE, CAST_TABLE, FUNC_TABLE};
 use crate::sql_gen::{SqlGenerator, SqlGeneratorContext};
 
-static FUNC_TABLE: LazyLock<HashMap<DataTypeName, Vec<FuncSign>>> = LazyLock::new(|| {
-    let mut funcs = HashMap::<DataTypeName, Vec<FuncSign>>::new();
-    func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
-    funcs
-});
-
-static AGG_FUNC_TABLE: LazyLock<HashMap<DataTypeName, Vec<AggFuncSig>>> = LazyLock::new(|| {
-    let mut funcs = HashMap::<DataTypeName, Vec<AggFuncSig>>::new();
-    agg_func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
-    funcs
-});
-
-/// Build a cast map from return types to viable cast-signatures.
-/// NOTE: We avoid cast from varchar to other datatypes apart from itself.
-/// This is because arbitrary strings may not be able to cast,
-/// creating large number of invalid queries.
-static CAST_TABLE: LazyLock<HashMap<DataTypeName, Vec<CastSig>>> = LazyLock::new(|| {
-    let mut casts = HashMap::<DataTypeName, Vec<CastSig>>::new();
-    cast_sigs()
-        .filter(|cast| {
-            cast.context == CastContext::Explicit || cast.context == CastContext::Implicit
-        })
-        .filter(|cast| {
-            cast.from_type != DataTypeName::Varchar || cast.to_type == DataTypeName::Varchar
-        })
-        .for_each(|cast| casts.entry(cast.to_type).or_default().push(cast));
-    casts
-});
-
```

Comment:
Refactored data structures which capture type signature information into the `types` module.